docs/development/new-cloud-provider.md

```diff
@@ -0,0 +1,65 @@
+# Adding Cloud Providers
+
+This document provides an overview of how to integrate a new cloud provider into Gardener. Each component that requires integration has a detailed description of how to integrate it and the steps required.
+
+## Cloud Components
+
+Gardener is composed of 2 or more Kubernetes clusters:
+
+* Shoot: These are one more "workload" clusters, the regular Kubernetes clusters you have seen. They provide places for your workloads to run.
+* Seed: This is the "management" cluster. It provides management service for one or more Shoot cluster.
```

Comment:
"Management service" is quite general, let's clarify that: It manages the control planes of multiple shoot clusters by running them as native Kubernetes workload.